### ISSUE TITLE
enhancement

### ISSUE THREAD

**Issue by christiancadieux2** [xaksis/vue-good-table#191]
an optional second table row that takes all the columns (skip the first column) to display things like long text (errors for example) that dont fit  well in a column and are not always present. so lets say secondary='error' , then display a separate row if error !== '' . that field takes all columns (skip the first for indentation) . thanks

**Comment by xaksis** [comment]
@christiancadieux2 any reason this error notification shouldn't be outside the table?

**Comment by christiancadieux2** [comment]
error was an example. but in this case the error information is different on each row and is easier to display right under each row as a separate row, sometimes with a + sign on the row to expand the row if applicable. I guess my request if for the ability to have 2 or more rows in the template with v-if on the rows, but right now, the template can only generate one row, when I try to include </tr><tr> in the template to generate a second row, it fails. maybe I am not doing it correctly, maybe I just need to use <span v-if='error' v-html="data"></span> and it would work...

**Comment by christiancadieux2** [comment]
correction: maybe i just need to use < .  span v-html="rawHtml"   >< .  / span .  >

**Comment by christiancadieux2** [comment]
Example: 
![expand](https://user-images.githubusercontent.com/31523474/36277936-888605aa-124f-11e8-8ecf-54a773e3a5bc.png)

**Comment by christiancadieux2** [comment]
tried the 'span' idea but it does not work, need a second template slot that can use v-if ...

**Comment by christiancadieux2** [comment]
multiple templates with v-if does not work, I guess the condition is not resolved per-row. so 2 templates with different slot and a new attribute to control when to display the second slot could work.

**Comment by xaksis** [comment]
@ah yeah, Unfortunately vue-good-table doesn't support that yet. There is an enhancement ticket for total/grouping of rows but there's no ETA for that yet.

**Comment by christiancadieux2** [comment]
I was imagining something simplistic like this in Table.vue: 
![extra](https://user-images.githubusercontent.com/31523474/36281714-dd2a4d6c-125b-11e8-8e30-9540c6dae123.png)

maybe I can play with it.
thanks for the quick response.

**Comment by christiancadieux2** [comment]
the following is sufficient to make it work with optional row:
<tbody>
          <template v-for="(row, index) in paginated">
          <tr
            :key="index"
            :class="getRowStyleClass(row)"
            @click="click(row, index)">
            <th v-if="lineNumbers" class="line-numbers">{{ getCurrentIndex(index) }}</th>
            <slot name="table-row-before" :row="row" :index="index"></slot>
            <slot name="table-row" :row="row" :formattedRow="formattedRow(row)" :index="index">
              <td
                v-for="(column, i) in columns"
                :key="i"
                :class="getClasses(i, 'td')"
                v-if="!column.hidden && column.field">
                <span v-if="!column.html">{{ collectFormatted(row, column) }}</span>
                <span v-if="column.html" v-html="collect(row, column.field)"></span>
              </td>
            </slot>
            <slot name="table-row-after" :row="row" :index="index">
            </slot>
          </tr>
          <tr v-if="row._extraRow">
            <slot name="table-extra-row" :row="row"  :index="index">
            </slot>
          </tr>
          </template>
          <tr v-if="processedRows.length === 0">
            <td :colspan="columns.length">
              <slot name="emptystate">
                <div class="center-align text-disabled">
                  No data for table.
                </div>
              </slot>
            </td>
          </tr>
        </tbody>

and the application look like this with an extra template:
<template slot="table-extra-row" slot-scope="props">
            <td></td>
            <td v-if="props.row.error" colspan="20">
              {{props.row.error }}
            </td>
          </template>

right now it's hardcored to add the extra row when _extraRow=1 in the data of that special row. the extra row content can be anything in the template...
thanks

**Comment by xaksis** [comment]
@christiancadieux2 while custom solutions like these might work in the short term, over time it'll make the component a lot less scalable and maintainable. Having said that, there is clearly a need for a more customized table with extra rows/summary rows/grouping etc. I'll have to plan out the exact approach for this. I'll keep this enhancement request open for now.
